### PR TITLE
[[ Bug 18277 ]] Translate NSRect <-> MCRectangle correctly

### PR DESCRIPTION
**Pull request by montegoulding**

### CODE REVIEW

File: engine/src/mac-core.mm [hunk 1]
```diff
@@ -1949,49 +1937,51 @@ MCPlatformModifiers MCMacPlatformMapNSModifiersToModifiers(NSUInteger p_modifier
...
+
 void MCMacPlatformMapScreenMCPointToNSPoint(MCPoint p, NSPoint& r_point)
 {
-	if (!s_have_primary_screen_height)
-	{
-		MCRectangle t_viewport;
-		MCPlatformGetScreenViewport(0, t_viewport);
-	}
+	if (!s_have_desktop_height)
+        get_desktop_height();
```
Comment (peter-b) [inline engine/src/mac-core.mm]: I suggest you make `get_desktop_height()` return the desktop height, and make that the only function that accesses the static variables.  Then, for example:

``` c++
void MCMacPlatfromMapScreenMCPointToNSPoint(MCPoint p, NSPoint &r_point)
{
    r_point = NSMakePoint(p.x, get_desktop_height() - p.y);
}
```

**Review by peter-b — CHANGES_REQUESTED** [review]

File: engine/src/mac-core.mm (hunk 1 above)
Comment (montegoulding) [reply engine/src/mac-core.mm]: OK, will do

**Comment by montegoulding** [comment]
@peter-b Hmm... I had this the other day with Travis timing out. Retrying.

**Comment by montegoulding** [comment]
@peter-b it seems the retry worked  ¯_(⊙︿⊙)_/¯

**Review by peter-b — APPROVED** [review]
This seems like a reasonable approach.  Do you think caching this data could cause any problems if the screen configuration changes while LiveCode is running?

File: engine/src/mac-core.mm [hunk 2]
```diff
@@ -512,7 +512,7 @@ - (void)applicationDidUpdate:(NSNotification *)notification
 - (void)applicationDidChangeScreenParameters:(NSNotification *)notification
 {
 	// Make sure we refetch the primary screen height.
-	s_have_primary_screen_height = false;
+	s_have_desktop_height = false;
```
Comment (montegoulding) [inline engine/src/mac-core.mm:515]: @peter-b this ensures that it will be updated after a screen config change

**Comment by peter-b** [comment]
Looks good to me!
@livecode-vulcan review ok ef1fc00

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok ef1fc0034a9cd62a3226fba22c626dd9693087ea

**Comment by montegoulding** [comment]
@peter-b before this is merged I'm just wondering if it should be retargeted to `develop-8.1`

**Comment by peter-b** [comment]
Yes that's probably a good idea.

**Comment by peter-b** [comment]
@livecode-vulcan review ok f6046d8

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok f6046d85c4575b128d4436c3221b28b8f32110e9

**Comment by livecode-vulcan** [comment]
😎 test success f6046d85c4575b128d4436c3221b28b8f32110e9

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/640)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/645)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/640)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/644)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/649)
* try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/641)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/641)

**Pull request merged**